pkg/gofr/service/retry_test.go

```diff
@@ -0,0 +1,224 @@
+package service
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"gofr.dev/pkg/gofr/logging"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockHTTP struct{}
+
+func (m *mockHTTP) HealthCheck(_ context.Context) *Health {
+	return &Health{
+		Status:  "UP",
+		Details: map[string]interface{}{"host": "http://test.com"},
+	}
+}
+
+func (m *mockHTTP) getHealthResponseForEndpoint(_ context.Context, _ string) *Health {
+	return &Health{
+		Status:  "UP",
+		Details: map[string]interface{}{"host": "http://test.com"},
+	}
+}
+
+func (m *mockHTTP) Get(_ context.Context, _ string, _ map[string]interface{}) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) GetWithHeaders(_ context.Context, _ string, _ map[string]interface{}, _ map[string]string) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) Post(_ context.Context, _ string, _ map[string]interface{}, _ []byte) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusCreated, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) PostWithHeaders(_ context.Context, _ string, _ map[string]interface{}, _ []byte,
+	_ map[string]string) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusCreated, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) Put(_ context.Context, _ string, _ map[string]interface{}, _ []byte) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) PutWithHeaders(_ context.Context, _ string, _ map[string]interface{}, _ []byte,
+	_ map[string]string) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) Patch(_ context.Context, _ string, _ map[string]interface{}, _ []byte) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) PatchWithHeaders(_ context.Context, _ string, _ map[string]interface{}, _ []byte,
+	_ map[string]string) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusOK, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) Delete(_ context.Context, _ string, _ []byte) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusNoContent, Body: http.NoBody}, nil
+}
+
+func (m *mockHTTP) DeleteWithHeaders(_ context.Context, _ string, _ []byte, _ map[string]string) (*http.Response, error) {
+	return &http.Response{StatusCode: http.StatusNoContent, Body: http.NoBody}, nil
+}
+
+func TestRetryProvider_Get(t *testing.T) {
+	mockHTTP := &mockHTTP{}
+	retryConfig := &RetryConfig{MaxRetries: 3}
+	retryHTTP := retryConfig.AddOption(mockHTTP)
+
+	// Make the GET request
+	resp, err := retryHTTP.Get(context.Background(), "/test", nil)
+	assert.Nil(t, err)
```

Comment:
Here and everywhere

```suggestion
	assert.NoError(t, err)
```

You can add `testifylint` to golangci-lint linter